ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/TransactionContractKeysValidation.scala

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils.committer.transaction
+
+import com.daml.ledger.participant.state.kvutils.Conversions
+import com.daml.ledger.participant.state.kvutils.DamlKvutils.{
+  DamlContractKey,
+  DamlStateKey,
+  DamlStateValue,
+}
+import com.daml.ledger.participant.state.kvutils.committer.Committer.Step
+import com.daml.ledger.participant.state.kvutils.committer.transaction.TransactionCommitter.{
+  DamlTransactionEntrySummary,
+  damlContractKey,
+  globalKey,
+}
+import com.daml.ledger.participant.state.kvutils.committer.{StepContinue, StepResult}
+import com.daml.ledger.participant.state.v1.RejectionReason
+import com.daml.lf.data.Ref.TypeConName
+import com.daml.lf.data.Time.Timestamp
+import com.daml.lf.transaction.{Node, NodeId}
+import com.daml.lf.value.Value
+import com.daml.lf.value.Value.ContractId
+
+object TransactionContractKeysValidation {
+
+  private type RawContractId = String
```

Comment:
Slightly more readable than `String` (as it was before) but not more type safe. I considered using `ContractId` but it's probably overkill in this context.